File: client/src/interfaces/LetterState.ts
import type { Letter } from './Letter';
import type { User } from './User';

export interface LetterState
{
    id: string;
    letterId: string;
    letter: Letter;
    userId: string;
    user: User;
    starred: boolean;
    isDeleted: boolean;
    isRead: boolean;
}

File: Domain/Models/LetterState.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Domain.Models
{
    public class LetterState
    {
        public Guid Id { get; set; }
        public Guid LetterId { get; set; }
        public Letter Letter { get; set; }
        public Guid UserId { get; set; }
        public User User { get; set; }
        public bool Starred { get; set; }
        public bool IsDeleted { get; set; }
        public bool IsRead { get; set; }

        public LetterState() 
        {
            Id = Guid.NewGuid();
            IsDeleted = false;
            Starred = false;
        }

    }
}
